components/restapi/core/blocks.go

```diff
@@ -121,16 +120,7 @@ func sendBlock(c echo.Context) (*api.BlockCreatedResponse, error) {
 
 	blockID, err := deps.BlockHandler.AttachBlock(c.Request().Context(), iotaBlock)
 	if err != nil {
-		switch {
-		case ierrors.Is(err, blockhandler.ErrBlockAttacherInvalidBlock):
-			return nil, ierrors.Wrapf(httpserver.ErrInvalidParameter, "failed to attach block: %w", err)
-
-		case ierrors.Is(err, blockhandler.ErrBlockAttacherAttachingNotPossible):
-			return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to attach block: %w", err)
-
-		default:
-			return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to attach block: %w", err)
-		}
+		return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to attach block: %w", err)
```

Comment:
Should really every error that could potentially happen be returned as 500 internal server error?
I think some of the possible errors should still be user errors, which should return `http.StatusBadRequest` 400 error, for example like the `ErrInvalidBlockVersion`, `ErrBlockTimeTooFarAheadInFuture`, `ErrValidatorNotInCommittee`, `ErrBlockParentNotFound`, `ErrBlockIssuingTimeNonMonotonic` etc...

Reply:
We should do this in another PR maybe in the API package that also contains the caches. I will create an issue.

I updated [#752](https://github.com/iotaledger/iota-core/issues/752) to add this.